src/client.ts

```diff
@@ -235,6 +239,32 @@ export default class Client implements ClientInterface {
     this.#ruby = ruby;
   }
 
+  async determineFormatter() {
+    const configuration = vscode.workspace.getConfiguration("rubyLsp");
+    const configuredFormatter: string = configuration.get("formatter")!;
+    switch (configuredFormatter) {
+      case "auto":
+        if (await this.projectHasDependency("rubocop")) {
+          this.client!.clientOptions.initializationOptions.formatter =
+            "rubocop";
+          this.client!.restart();
+          return "rubocop";
+        } else if (await this.projectHasDependency("syntax_tree")) {
+          this.client!.clientOptions.initializationOptions.formatter =
```

Comment:
This is not necessary. We're already setting the initialization option in line 119. Whatever is configured, we send to the server. The server will have to decide which formatter to use.

The code on the client should just decide which information to display on the status item, not change what we send the server.

Reply:
Got it. I was mistakenly thinking that this would act the 'source of truth' to determine what formatter to use, but instead we can just the gem mirror this behaviour.